libraries/rush-lib/src/logic/pnpm/PnpmProjectManifestConfiguration.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import { JsonFile, JsonSchema, FileSystem } from '@rushstack/node-core-library';
+
+/**
+ * Pnpm config in package.json
+ * @beta
+ */
+export interface IPnpmProjectManifestConfigurationJson {
```

Comment:
@iclanton and I have talked about breaking the PNPM configs out into their own `pnpm-config.json` file like this PR is now doing. Maybe this should be made into a more generic name? `IPnpmConfigurationJson` and the class `PnpmConfiguration` (and same for all similarly named vars)? This might be a good time to do that.